plugins/rebalance/process.go

```diff
@@ -0,0 +1,110 @@
+package rebalance
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os/exec"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	rebalanceapi "github.com/gluster/glusterd2/plugins/rebalance/api"
+
+	"github.com/cespare/xxhash"
+	config "github.com/spf13/viper"
+)
+
+const (
+	glusterfsBin = "glusterfs"
+)
+
+// Process type represents information about rebalance process
+type Process struct {
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+	rInfo          rebalanceapi.RebalInfo
+}
+
+// Name returns the process name
+func (r *Process) Name() string {
+	return "rebalance"
+}
+
+// Path returns absolute path to the binary of rebalance process
+func (r *Process) Path() string {
+	return r.binarypath
+}
+
+// Args returns arguments to be passed to rebalance process
+func (r *Process) Args() string {
+
+	volfileserver, port, _ := net.SplitHostPort(config.GetString("clientaddress"))
+	if volfileserver == "" {
+		volfileserver = "localhost"
+	}
+
+	volFileID := path.Join("rebalance/", r.rInfo.Volname)
+	logDir := path.Join(config.GetString("logdir"), "glusterfs")
+	logFile := fmt.Sprintf("%s/%s-rebalance.log", logDir, r.rInfo.Volname)
+	cmd := r.rInfo.Cmd
+	commithash := r.rInfo.CommitHash
+
+	var buffer bytes.Buffer
```

Comment:
need to create vector args instead of a string,  if  #781 gets merged

Reply:
I will wait until it is merged before I do that